core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -3027,6 +3028,43 @@ private void checkSarg(String message, Sarg sarg,
         is(false));
   }
 
+  private void helpTestFloorCeil(String timestampString, String expectedFloorString,
+      String expectedCeilString, TimeUnitRange timeUnitRange) {
+    final RexLiteral timestampLiteral = rexBuilder.makeTimestampLiteral(
+        new TimestampString(timestampString), 3);
+    final RexLiteral flagLiteral = rexBuilder.makeFlag(timeUnitRange);
+
+    RexNode flooredLiteral = rexBuilder.makeCall(SqlStdOperatorTable.FLOOR,
+        ImmutableList.of(timestampLiteral, flagLiteral));
+    long expectedFloorValue = new TimestampString(expectedFloorString).getMillisSinceEpoch();
+    assertThat(eval(flooredLiteral), is(expectedFloorValue));
+
+    RexNode ceiledLiteral = rexBuilder.makeCall(SqlStdOperatorTable.CEIL,
```

Comment:
Please be consistent with the `final` modifier, since you have added it on some other vars, if you omit it here, the reader thinks it's going to mutate, while this is not the case

Reply:
added final modifier here